metrics-operator/controllers/common/providers/dynatrace/dynatrace_test.go

```diff
@@ -351,13 +361,15 @@ func TestEvaluateQuery_HappyPath(t *testing.T) {
 			TargetServer: svr.URL,
 		},
 	}
-	r, raw, e := kdp.EvaluateQuery(context.TODO(), obj, p)
-	require.Nil(t, e)
-	require.Equal(t, []byte(dtpayload), raw)
-	require.Equal(t, fmt.Sprintf("%f", 50.0), r)
+	for _, obj := range objects {
+		r, raw, e := kdp.EvaluateQuery(context.TODO(), obj, p)
+		require.Nil(t, e)
+		require.Equal(t, []byte(dtpayload), raw)
+		require.Equal(t, fmt.Sprintf("%f", 50.0), r)
+	}
 }
 
-func setupTest(objs ...client.Object) (KeptnDynatraceProvider, metricsapi.KeptnMetric) {
+func setupTest(objs ...client.Object) (KeptnDynatraceProvider, [2]metricsapi.KeptnMetric) {
```

Comment:
I would keep it dynamic so we don't have to update this field each time we add an entry
```suggestion
func setupTest(objs ...client.Object) (KeptnDynatraceProvider, []metricsapi.KeptnMetric) {
```